idp_task1.cpp

```diff
@@ -4,52 +4,59 @@
 static const int LISTENQ = 16;
 static const int DEFAULT_ERROR = -1;
 static const int SERVER_PORT = 8877;
-static const int ECHO_STR_LEN = 6;                                            
+static const char* echo_str_prefix = "echo: ";
+static const int ECHO_STR_LEN = strlen(echo_str_prefix);
 
 void createTestTCPServer()
 {
     try 
     {
-    std::unique_ptr<SocketWrapper> server_socket = std::make_unique<SocketWrapper>(AF_INET, SOCK_STREAM, 0, SERVER_PORT);
+        SocketWrapper_uptr server_socket = std::make_unique<SocketWrapper>(AF_INET, SOCK_STREAM, 0, SERVER_PORT);
 
-    server_socket->Bind();
+        sockaddr_in server_socket_address;
+        bzero(&server_socket_address, sizeof(sockaddr_in));
+        server_socket_address.sin_family = AF_INET;
+        server_socket_address.sin_port = htons (SERVER_PORT);
+        server_socket_address.sin_addr.s_addr = INADDR_ANY ; 
 
-    server_socket->Listen(LISTENQ);
+        server_socket->Bind(&server_socket_address);
 
-    while (true) 
-    {
-		std::unique_ptr<SocketWrapper> client_socket = server_socket->Accept();
+        server_socket->Listen(LISTENQ);
+
+        while (true) 
+        {
+            SocketWrapper_uptr client_socket = server_socket->Accept();
 
-		int n = 0;
-		int maxlen = 100;
-		char buffer[maxlen];
+            int n = 0;
+            int maxlen = 100;
+            char buffer[maxlen];
 
-        char echo[maxlen + ECHO_STR_LEN];
-        strcpy(echo, "echo: \0");
-        memset(buffer, 0, maxlen);
+            char echo[maxlen + ECHO_STR_LEN];
+            strcpy(echo, echo_str_prefix);
+            memset(buffer, 0, maxlen);
 
-		printf("client connected with ip address: %s\n",
+            printf("client connected with ip address: %s\n",
 		       inet_ntoa(server_socket->GetSockAddrStruct()->sin_addr));
 
-		while ((n = recv(client_socket->GetSocketHandle(), buffer, maxlen, 0)) > 0)
-        {
-			printf("received: %s", buffer);
+            while (client_socket->ReceiveData(buffer, maxlen, 0) > 0)
```

Comment:
How do we know amount of data received from socket here? It's not critical for you now, but one day you'll need to know it anyway.